### PR TITLE
selectdefaultapplication: init at unstable-2021-08-12

### LINKED ISSUE #174133: Packaging request: selectdefaultapplication
**Project description**

Selectdefaultapplication is a very simple GUI application that lets you define default applications on Linux in a sane way.
Basically it just loads all installed applications by reading their .desktop files, reads the MimeType fields to see what it supports, and updates ~/.config/mimeapps.list with what the user wants.
Not all desktop environments provide such a tool, prompting for an agnostic tool.

**Metadata**

* homepage URL: https://github.com/sandsmark/selectdefaultapplication
* source URL: https://github.com/sandsmark/selectdefaultapplication
* license: gpl2
* platforms: linux

**Additional info:**
AUR package: https://aur.archlinux.org/packages/selectdefaultapplication-git

### PR DESCRIPTION
**Pull request by peeley**
###### Description of changes

Creates the package definition for `selectdefaultapplication`, an application to select default applications for various file types on Linux systems.

Also adds myself (nsnelson) to the maintainer list and as the maintainer of `selectdefaultapplication`.

Closes #174133.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14625,4 +14625,10 @@
     github = "snpschaaf";
     githubId = 105843013;
   };
+  nsnelson = {
+    email = "noah.snelson@protonmail.com";
+    name = "Noah Snelson";
+    github = "peeley";
+    githubId = 30942198;
+  };
```
Comment (muscaln) [inline maintainers/maintainer-list.nix:14633]: This should be done in a seperate commit.

File: pkgs/applications/misc/selectdefaultapplication/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ mkDerivation, lib, fetchFromGitHub, qmake, hicolor-icon-theme, qtbase }:
```
Comment (muscaln) [inline pkgs/applications/misc/selectdefaultapplication/default.nix:1]:
```suggestion
{ stdenv, lib, fetchFromGitHub, qmake, qtbase }:
```

File: pkgs/applications/misc/selectdefaultapplication/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+mkDerivation {
```
Comment (muscaln) [inline pkgs/applications/misc/selectdefaultapplication/default.nix:3]:
```suggestion
stdenv.mkDerivation {
```

File: pkgs/applications/misc/selectdefaultapplication/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    owner = "sandsmark";
+    repo = "selectdefaultapplication";
+    rev = "c752df6ba8caceeef54bcf6527f1bccc2ca8202a";
+    sha256 = "C/70xpt6RoQNIlAjSJhOCyheolK4Xp6RiSZmeqMP4fw=";
+  };
+
+  nativeBuildInputs = [
+    qmake
+    qtbase
+  ];
```
Comment (muscaln) [inline pkgs/applications/misc/selectdefaultapplication/default.nix:17]:
```suggestion
  nativeBuildInputs = [ qmake ];
  buildInputs = [ qtbase ];
```

**Review by muscaln — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (peeley) [reply maintainers/maintainer-list.nix:14633]: Removed myself and then re-added in a separate commit. Not sure if I should have just made a separate PR, first time contributing!

**Comment by peeley** [comment]
@ofborg eval

**Comment by peeley** [comment]
@ofborg eval

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -10440,7 +10440,10 @@ with pkgs;
 
   seexpr = callPackage ../development/compilers/seexpr { };
 
+  selectdefaultapplication = libsForQt5.callPackage ../applications/misc/selectdefaultapplication { };
+
   semgrep = python3.pkgs.callPackage ../tools/security/semgrep { };
+
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:10446]: Those are very closely related that's why there is no new line between them.

**Review by SuperSandro2000 — COMMENTED** [review]
Please squash all the review committs into the init commit and keep the extra commit for the maintainer addition.

File: pkgs/top-level/all-packages.nix (hunk 5 above)
Comment (peeley) [reply pkgs/top-level/all-packages.nix:10446]: Removed!

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -14720,4 +14720,10 @@
     github = "dfithian";
     githubId = 8409320;
   };
+  nsnelson = {
+    email = "noah.snelson@protonmail.com";
+    name = "Noah Snelson";
+    github = "peeley";
+    githubId = 30942198;
+  };
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:14728]: Please sort this alphabetical

Comment (peeley) [reply maintainers/maintainer-list.nix]: Done!

Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix]: you reverted it when force pushing

Comment (peeley) [reply maintainers/maintainer-list.nix]: Whoops, my bad! For some reason, I thought you meant alphabetizing the fields of the `nsnelson` map 😅. Should be fixed now.

**Pull request merged**